### PR TITLE
Feature/2843 refactor logging

### PR DESCRIPTION
**Pull request by ArturFriesen**
## Description
This PR refactors logging to use os_log.
The logs for the debug menu are saved into a file instead of the NSUserDefaults.
Its now possible to select between log types at the debug menu.

Simulator Screen Shot - iPhone 8 - 2020-10-08 at 14 09 26

Simulator Screen Shot - iPhone 8 - 2020-10-08 at 14 09 08

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Workers/Logging.swift [hunk 1]
```diff
@@ -0,0 +1,149 @@
+import Foundation
+import os.log
+
+extension OSLog {
+
+	private static var subsystem = Bundle.main.unwrappedBundleIdentifier
+
+    /// API interactions
+    static let api = OSLog(subsystem: subsystem, category: "api")
+    /// Exoplanet UI
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Workers/Logging.swift:10]: Exoplanet? 😁

File: src/xcode/ENA/ENA/Source/Workers/Logging.swift [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    static func warning(_ message: String, log: OSLog = .default) {
+        Self.log(message: message, type: .default, log: log, error: nil)
+    }
+
+    static func error(_ message: String, log: OSLog = .default, error: Error? = nil) {
+        Self.log(message: message, type: .error, log: log, error: error)
+    }
+
+	private static func log(message: String, type: OSLogType, log: OSLog, error: Error?) {
+		os_log("%{private}@", log: log, type: type, message)
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Workers/Logging.swift:35]: AFAIK we are not allowed to even to os_log in Production

**Review by marcussc — CHANGES_REQUESTED** [review]
Very nice PR, only got two small remarks

File: src/xcode/ENA/ENA/Source/Workers/Logging.swift (hunk 1 above)
Comment (whiskey) [reply src/xcode/ENA/ENA/Source/Workers/Logging.swift:10]: YES! :D

File: src/xcode/ENA/ENA/Source/Workers/Logging.swift (hunk 2 above)
Comment (whiskey) [reply src/xcode/ENA/ENA/Source/Workers/Logging.swift:35]: But everything is "private", i.e. log output of the whole `message` will be:
`DATE_FOO <private>`

Comment (marcussc) [reply src/xcode/ENA/ENA/Source/Workers/Logging.swift:35]: What I am wondering currently, how do we differentiate between TestFlight and AppStore builds?

Comment (whiskey) [reply src/xcode/ENA/ENA/Source/Workers/Logging.swift:35]: `RELEASE` seems to be set for release builds. So:

```
#if !RELEASE
// log log log!
#endif
```

Comment (marcussc) [reply src/xcode/ENA/ENA/Source/Workers/Logging.swift:35]: Wouldn't this be the same for TestFlight Builds?

Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Workers/Logging.swift:37]: TestFlight builds are build with "TestFlight" configuration. And this configuration does not have a RELEASE flag configured in the build settings. So my guess is, logs would work for TestFlight releases.

Bildschirmfoto 2020-10-09 um 10 01 25

Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Workers/Logging.swift:37]: I have "hidden" os_log call behind !RELEASE

**Review by marcussc — APPROVED** [review]

**Review by whiskey — APPROVED** [review]

**Pull request merged**